geniza/corpus/tests/test_corpus_templatetags.py

```diff
@@ -161,3 +161,14 @@ def test_shelfmark_wrap():
         corpus_extras.shelfmark_wrap("foo + bar + baz")
         == "<span>foo</span> + <span>bar</span> + <span>baz</span>"
     )
+
+
+def test_translate_url(document):
+    # should translate requested URL into Hebrew
+    ctx = {"request": HttpRequest()}
+    ctx["request"].path = document.get_absolute_url()
+    assert corpus_extras.translate_url(ctx, "he").startswith("/he/")
+
+    # if a Hebrew version cannot be determined, should return the original URL
+    ctx["request"].path = "https://example.com"
+    assert corpus_extras.translate_url(ctx, "he") == "https://example.com"
```

Comment:
what are the cases where we can't get a translated version of the url? 

I don't suppose this would have any awareness of wagtail pages and whether a page is available in translation or not?

It seems like it would be nice to disable the switcher when a page can't actually be switched, but that might need to be out of scope for now

Reply:
> what are the cases where we can't get a translated version of the url?

Basically I was just testing generic "bad input" to this template tag and not really thinking through a more realistic use case. In the current implementation, it'll never be the case! (At least, I think all paths within the app are translatable.) Can remove the test if we can't think of any real world examples.

> I don't suppose this would have any awareness of wagtail pages and whether a page is available in translation or not?

Right, it's just relying on Django [`translate_url`](https://docs.djangoproject.com/en/3.2/_modules/django/urls/base/), so if the slug gets translated it won't work (as in #582). Probably need to do something [like this](https://github.com/wagtail/wagtail-localize/discussions/308#discussioncomment-303188) instead. But maybe it makes sense to roll #582 in with this issue?

> It seems like it would be nice to disable the switcher when a page can't actually be switched, but that might need to be out of scope for now

I agree on both.